refactor node view

pleimer opened the pull request:
Here's a re factor and consolidation of the infrastructure node view dashboard. This gets rid of a lot of the redundant panels that existed in the old version as well as the panels describing the characteristics of the system (total CPU cores, total memory, and the always wrong total disk size). IMO, this information is not all that useful to monitoring node performance and just takes up space. The new `Quickview` section contains useful information at a glance, and the following sections dig deeper into their respective categories. On the interface panels, Rx and Tx directions are combined into a single panel, with Tx values showing up as a negative value. This allows more information to be displayed with less space. 

Additionally, all attempts to calculate disk size based on the size of the mounted file systems have been thrown out for two reasons:
1. It's nearly impossible to build a query that can figure out hardware disk size based on file systems in all environments
2. The information is not useful.
It is more useful to measure how much of each partition is being used. Thus, I replaced all attempts to determine how much of the disk is used with how much of each file system is used. This fixes https://bugzilla.redhat.com/show_bug.cgi?id=1843667

![image](https://user-images.githubusercontent.com/41337120/99824809-34d31d80-2b24-11eb-947b-4c0e387da334.png)

leifmadsen commented:
First of all, this looks really good!

Couple comments inline.

> Here's a re factor and consolidation of the infrastructure node view dashboard. This gets rid of a lot of the redundant panels that existed in the old version as well as the panels describing the characteristics of the system (total CPU cores, total memory, and the always wrong total disk size). IMO, this information is not all that useful to monitoring node performance and just takes up space. 

Yea I can get behind that logic.

> The new `Quickview` section contains useful information at a glance, and the following sections dig deeper into their respective categories. 

Yes! I had always wanted all those gauges moved up since I think seeing that information together provides value.

> On the interface panels, Rx and Tx directions are combined into a single panel, with Tx values showing up as a negative value. This allows more information to be displayed with less space.

Nice!

> Additionally, all attempts to calculate disk size based on the size of the mounted file systems have been thrown out for two reasons:
> 
>     1. It's nearly impossible to build a query that can figure out hardware disk size based on file systems in all environments
> 
>     2. The information is not useful.
>        It is more useful to measure how much of each partition is being used. Thus, I replaced all attempts to determine how much of the disk is used with how much of each file system is used. This fixes https://bugzilla.redhat.com/show_bug.cgi?id=1843667

Cool, that makes sense. The panels look so much better. The only question I have is the use of legends on some panels and no legends on others. Do you have some sort of workflow that determines when to enable the legend vs hiding it? I do like the cleaner overall look of the no-legend approach.

This looks to be much improved though!

leifmadsen commented on deploy/rhos-dashboard.yaml:
```diff
@@ -3056,7 +2225,7 @@ spec:
             "allValue": null,
             "current": {
-              "selected": false,
+              "selected": true,
```
I know it's existing, but some concerns about hard coding some of the default selections in to the dashboard itself. I feel like I addressed this in another dashboard though... but it's been a while.

pleimer replied:
Good point, removing the saved selection results in the first value being used. We just have to remember to do that on every update now.

leifmadsen replied:
I guess this is why things like grafyaml exist :)

csibbitt commented:
This all looks pretty good to me! I also noticed the legend inconsistency and would prefer to see them on every panel that has more than one line ("Data" and "Aggr. Usage by type" need them for sure. "Hugepages" probably does not need it.). I'd also suggest try moving them to the bottom instead of the right and see if they are still readable; doing that turns whitespace into graph space.

csibbitt commented on deploy/rhos-dashboard.yaml:
```diff
@@ -2394,156 +1540,165 @@ spec:
           "targets": [
             {
-              "expr": "sum(collectd_df_df_complex{plugin_instance!~\"devtmpfs|overlay|shm|tmpfs\",type_instance!=\"free\",host=\"$hosts\"})/sum(collectd_df_df_complex{plugin_instance!~\"devtmpfs|overlay|shm|tmpfs\",host=\"$hosts\"})",
+              "expr": "sum(collectd_hugepages_vmpage_number{type_instance=\"free\",host=\"$hosts\"})",
```
This appears to have changed from "type_instance=used" to "type_instace=free", which is not made clear in the title, legend, or labels. To be useful I think the graph needs to represent both the number being used, and what % that is of the total hugepages, if possible.

pleimer replied:
Im amazed you noticed that just by looking through the json data. That should never have been changed. For the update, I am including a separate guage panel that will show percentage of hugepages used.

pleimer commented:
Fixed hugepages querry error @csibbitt pointed out, added panel showing percentage of hugepages used, and included keys in all panels that include more than one metric. Panel sizes have been adjusted so that the first three keys are viewable on a standard 13.5in monitor, and they are now included below the graphs. This makes each graph take up more space on the screen, but they values are now selectable.

![image](https://user-images.githubusercontent.com/41337120/99986053-b624ed00-2d7c-11eb-914b-3457f6f88bc8.png)

csibbitt commented:
Hugepages graph scale should be positive values only.

This is looking really good, though! Nice work, Paul.

leifmadsen commented:
Yea this looks great, nice work! I looked again and don't see anything else to comment on.
